src/promptflow/promptflow/executor/batch_engine.py

```diff
@@ -62,9 +62,18 @@ def _resolve_data(self, input_dirs: Dict[str, str]):
         result = {}
         for input_key, input_dir in input_dirs.items():
             input_dir = self._resolve_dir(input_dir)
-            file_data = load_data(input_dir)
-            resolve_multimedia_data_recursively(input_dir, file_data)
-            result[input_key] = file_data
+            result[input_key] = self._resolve_data_from_input_dir(input_dir)
+        return result
+
+    def _resolve_data_from_input_dir(self, input_dir: Path):
+        """Resolve input data from directory"""
+        result = []
+        if input_dir.is_file():
```

Comment:
This seems weird, consider renaming to `input_path` ?
None blocking, though

Reply:
Will update it in later PR.

Reply:
Updated.